data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/JacksonEvent.java

```diff
@@ -134,13 +138,20 @@ private JsonNode getOrCreateNode(final JsonNode node, final String key) {
 
         final String trimmedKey = checkAndTrimKey(key);
 
-        final JsonPointer jsonPointer = toJsonPointer(trimmedKey);
-
-        JsonNode node = jsonNode.at(jsonPointer);
+        final JsonNode node = getNode(trimmedKey);
```

Comment:
This is done because a key can start with or without a `/`. I found it easier to ensure the key was cleaned to avoid a redundant slash. The trimming also ensures there are no keys that end with a slash.